PIPELINE.md

```diff
@@ -0,0 +1,849 @@
+# An in-depth guide to the ALeRCE pipeline
+
+Every step works as a "event driven microservice", which listens to a topic and must receive one or more messages in order to perform their operations. The pipeline steps are developed under the [APF Framework](https://github.com/alercebroker/APF). This framework gives some strong guidelines on the structure of the step, in addition of providing some hooks which facilitates the parsing of the information received by the step and the further deployment as well.  
+
+## Pipeline Diagram (detailed)
+
+![Image](https://user-images.githubusercontent.com/20263599/229163793-f0cefe89-6a2b-4dee-a111-20da2eec3461.png)
+
+## Steps specification
+
+### Sorting Hat
+
+The Sorting Hat is a gateway step which assigns an ALeRCE ID (aid) to every incoming alert. This `aid` can be newly generated or an already existing one, depending on the result of the crossmatch performed on the object database. 
+At the present time, this step must be deployed once per survey, with different settings to work with the given survey. So you might find more than one instance of sorting hat deployed.
+
+#### Output Information
+
+```python
+{
+    "type": "record",
+    "doc": "Multi stream alert of any telescope/survey",
+    "name": "alerce.alert",
+    "fields": [
+        {"name": "oid", "type": "string"},
+        {"name": "tid", "type": "string"},
+        {"name": "pid", "type": "long"},
+        {"name": "candid", "type": ["long", "string"]},
+        {"name": "mjd", "type": "double"},
+        {"name": "fid", "type": "int"},
+        {"name": "ra", "type": "double"},
+        {"name": "dec", "type": "double"},
+        {"name": "mag", "type": "float"},
+        {"name": "e_mag", "type": "float"},
+        {"name": "isdiffpos", "type": "int"},
+        {"name": "e_ra", "type": "float"},
+        {"name": "e_dec", "type": "float"},
+        {
+            "name": "extra_fields",
+            "type": EXTRA_FIELDS,
+        },
+        {"name": "aid", "type": "string"},
+        {"name": "stamps", "type": STAMPS},
+    ],
+}
+```
+
+Given the **STAMPS** type:
+
+```python
+{
+    "type": "record",
+    "name": "stamps",
+    "fields": [
+        {"name": "science", "type": ["null", "bytes"], "default": None},
+        {"name": "template", "type": ["null", "bytes"], "default": None},
+        {"name": "difference", "type": ["null", "bytes"], "default": None},
+    ],
+}
+```
+
+and the **EXTRA_FIELDS** type:
+
+```python
+{
+    "type": "map",
+    "values": ["null", "int", "float", "string", "bytes"],
+    "default": {},
+}
+```
+
+### Stamp Classifier Step
+
+Applies a machine learning model to determine the classification of an object based on a image of the former, called stamp.
```

Comment:
```suggestion
Applies a machine learning model to determine the classification of an alert based on its stamps. This should have an instance per survey, since the ML models are survey dependent.
```